File: comscore/android/src/main/java/com/theoplayercomscore/integration/ComscoreStreamingAnalytics.java
package com.theoplayercomscore.integration;

import com.theoplayer.android.api.player.Player;

import java.util.Map;

public class ComscoreStreamingAnalytics {

    private ComscoreTHEOplayerAdapter adapter;

    public ComscoreStreamingAnalytics(Player player, String playerVersion, ComscoreConfiguration configuration, ComscoreMetaData metadata) {
        adapter = new ComscoreTHEOplayerAdapter(player, playerVersion, configuration,metadata);
    }

    public void reportEnd() {
        adapter.notifyEnd();
    }

    public void reportPause() {
        adapter.notifyPause();
    }

    public void setPersistentLabel(String label, String value) {
        adapter.setPersistentLabel(label,value);
    }

    public void setPersistentLabels(Map<String,String> labels) {
        adapter.setPersistentLabels(labels);
    }

    public void update(ComscoreMetaData metadata) {
        adapter.setMedatata(metadata);
    }
}

File: comscore/android/src/main/java/com/theoplayercomscore/integration/ComscoreStreamingAnalytics.kt
package com.theoplayercomscore.integration

import com.theoplayer.android.api.player.Player

class ComscoreStreamingAnalytics(
  player: Player,
  playerVersion: String?,
  configuration: ComscoreConfiguration,
  metadata: ComscoreMetaData
) {
  private val adapter: ComscoreTHEOplayerAdapter

  init {
    adapter = ComscoreTHEOplayerAdapter(player, playerVersion, configuration, metadata)
  }

  fun reportEnd() {
    adapter.notifyEnd()
  }

  fun reportPause() {
    adapter.notifyPause()
  }

  fun setPersistentLabel(label: String, value: String) {
    adapter.setPersistentLabel(label, value)
  }

  fun setPersistentLabels(labels: Map<String, String>) {
    adapter.setPersistentLabels(labels)
  }

  fun update(metadata: ComscoreMetaData) {
    adapter.setMedatata(metadata)
  }
}
